torch/distributions/exp_family.py

```diff
@@ -0,0 +1,74 @@
+import torch
+from torch.distributions.distribution import Distribution
+from torch.autograd import Variable
+
+
+class ExponentialFamily(Distribution):
+    r"""
+    ExponentialFamily is the abstract base class for probability distributions belonging to an
+    exponential family, whose form is defined below
+
+    ..math::
+
+        p_{F}(x; \theta) = \exp(\langle t(x), \theta\rangle) - F(\theta) + k(x))
+
+    where :math:`\theta` denotes the natural parameters, :math:`t(x)` denotes the sufficient statistic,
+    :math:`F(\theta)` is the log normalizer function for a given family and :math:`k(x)` is the carrier
+    measure.
```

Comment:
Could you please add:
1. A note that this is mainly used to test correctness of `.entropy()` and `kl_divergence()` and
2. A reference to the Bregman divergence paper?